## Preprocessing

In [ ]:
!pip install pyyaml h5py

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/


In [ ]:
# Import our dependencies
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import StandardScaler
import pandas as pd
import tensorflow as tf

#  Import and read the charity_data.csv.
import pandas as pd 
# Import our input dataset
# This piece of code prompts to choose file
from google.colab import files
uploaded = files.upload()

import os
from tensorflow import keras

# After choosing file use this piece of code
import io
df = pd.read_csv(io.BytesIO(uploaded['charity_data.csv']))
df.head()

Saving charity_data.csv to charity_data (2).csv


EIN                                      NAME APPLICATION_TYPE  \
0  10520599              BLUE KNIGHTS MOTORCYCLE CLUB              T10   
1  10531628    AMERICAN CHESAPEAKE CLUB CHARITABLE TR               T3   
2  10547893        ST CLOUD PROFESSIONAL FIREFIGHTERS               T5   
3  10553066            SOUTHSIDE ATHLETIC ASSOCIATION               T3   
4  10556103  GENETIC RESEARCH INSTITUTE OF THE DESERT               T3   

        AFFILIATION CLASSIFICATION      USE_CASE  ORGANIZATION  STATUS  \
0       Independent          C1000    ProductDev   Association       1   
1       Independent          C2000  Preservation  Co-operative       1   
2  CompanySponsored          C3000    ProductDev   Association       1   
3  CompanySponsored          C2000  Preservation         Trust       1   
4       Independent          C1000     Heathcare         Trust       1   

      INCOME_AMT SPECIAL_CONSIDERATIONS  ASK_AMT  IS_SUCCESSFUL  
0              0                      N     5000              1  
1         1-9999                      N   108590              1  
2              0                      N     5000              0  
3    10000-24999                      N     6692              1  
4  100000-499999                      N   142590              1

In [ ]:
application_df = pd.read_csv('charity_data.csv')
application_df

EIN                                               NAME  \
0       10520599                       BLUE KNIGHTS MOTORCYCLE CLUB   
1       10531628             AMERICAN CHESAPEAKE CLUB CHARITABLE TR   
2       10547893                 ST CLOUD PROFESSIONAL FIREFIGHTERS   
3       10553066                     SOUTHSIDE ATHLETIC ASSOCIATION   
4       10556103           GENETIC RESEARCH INSTITUTE OF THE DESERT   
...          ...                                                ...   
34294  996009318              THE LIONS CLUB OF HONOLULU KAMEHAMEHA   
34295  996010315           INTERNATIONAL ASSOCIATION OF LIONS CLUBS   
34296  996012607                                PTA HAWAII CONGRESS   
34297  996015768  AMERICAN FEDERATION OF GOVERNMENT EMPLOYEES LO...   
34298  996086871                           WATERHOUSE CHARITABLE TR   

      APPLICATION_TYPE       AFFILIATION CLASSIFICATION      USE_CASE  \
0                  T10       Independent          C1000    ProductDev   
1                   T3       Independent          C2000  Preservation   
2                   T5  CompanySponsored          C3000    ProductDev   
3                   T3  CompanySponsored          C2000  Preservation   
4                   T3       Independent          C1000     Heathcare   
...                ...               ...            ...           ...   
34294               T4       Independent          C1000    ProductDev   
34295               T4  CompanySponsored          C3000    ProductDev   
34296               T3  CompanySponsored          C2000  Preservation   
34297               T5       Independent          C3000    ProductDev   
34298               T3       Independent          C1000  Preservation   

       ORGANIZATION  STATUS     INCOME_AMT SPECIAL_CONSIDERATIONS   ASK_AMT  \
0       Association       1              0                      N      5000   
1      Co-operative       1         1-9999                      N    108590   
2       Association       1              0                      N      5000   
3             Trust       1    10000-24999                      N      6692   
4             Trust       1  100000-499999                      N    142590   
...             ...     ...            ...                    ...       ...   
34294   Association       1              0                      N      5000   
34295   Association       1              0                      N      5000   
34296   Association       1              0                      N      5000   
34297   Association       1              0                      N      5000   
34298  Co-operative       1          1M-5M                      N  36500179   

       IS_SUCCESSFUL  
0                  1  
1                  1  
2                  0  
3                  1  
4                  1  
...              ...  
34294              0  
34295              0  
34296              0  
34297              1  
34298              0  

[34299 rows x 12 columns]

In [ ]:
# df attribute: access a single value for a row/column pair by label

application_df.at[1,"APPLICATION_TYPE"]

'T3'

In [ ]:
# df attribute: access a single value for a row/column pair by integer position
application_df.iat[1,0]

10531628

In [ ]:
# Drop the non-beneficial ID columns, 'EIN' and 'NAME'.
application_df = application_df.drop(['EIN','NAME'],axis = 1)
application_df.head(2)

APPLICATION_TYPE  AFFILIATION CLASSIFICATION      USE_CASE  ORGANIZATION  \
0              T10  Independent          C1000    ProductDev   Association   
1               T3  Independent          C2000  Preservation  Co-operative   

   STATUS INCOME_AMT SPECIAL_CONSIDERATIONS  ASK_AMT  IS_SUCCESSFUL  
0       1          0                      N     5000              1  
1       1     1-9999                      N   108590              1

Target: 'IS_SUCCESSFUL'
Features: 'APPLICATION_TYPE', 'AFFILIATION', 'CLASSIFICATION', 'USE_CASE',
       'ORGANIZATION', 'STATUS', 'INCOME_AMT', 'SPECIAL_CONSIDERATIONS',
       'ASK_AMT'

In [ ]:
application_df.dtypes

APPLICATION_TYPE          object
AFFILIATION               object
CLASSIFICATION            object
USE_CASE                  object
ORGANIZATION              object
STATUS                     int64
INCOME_AMT                object
SPECIAL_CONSIDERATIONS    object
ASK_AMT                    int64
IS_SUCCESSFUL              int64
dtype: object

In [ ]:
# Determine the number of unique values in each column.
application_cat = application_df.dtypes[application_df.dtypes =='object'].index.tolist()
application_df[application_cat].nunique()

APPLICATION_TYPE          17
AFFILIATION                6
CLASSIFICATION            71
USE_CASE                   5
ORGANIZATION               4
INCOME_AMT                 9
SPECIAL_CONSIDERATIONS     2
dtype: int64

In [ ]:
# Look at APPLICATION_TYPE value counts for binning
#  YOUR CODE GOES HERE

apptype = application_df['APPLICATION_TYPE'].value_counts()
apptype
#bin apptypes with counts <100 - would create 10 bins
#helper text indicaters binning with counts <200 - would create 9 bins
#try <600


T3     27037
T4      1542
T6      1216
T5      1173
T19     1065
T8       737
T7       725
T10      528
T9       156
T13       66
T12       27
T2        16
T25        3
T14        3
T29        2
T15        2
T17        1
Name: APPLICATION_TYPE, dtype: int64

In [ ]:
type(apptype)

pandas.core.series.Series

In [ ]:
attr = apptype.where(apptype<100)
attr.dropna(inplace = True)
attr

T13    66.0
T12    27.0
T2     16.0
T25     3.0
T14     3.0
T29     2.0
T15     2.0
T17     1.0
Name: APPLICATION_TYPE, dtype: float64

In [ ]:
attr2 = apptype.where(apptype<10)
attr2.dropna(inplace = True)
attr2

T25    3.0
T14    3.0
T29    2.0
T15    2.0
T17    1.0
Name: APPLICATION_TYPE, dtype: float64

In [ ]:
# Choose a cutoff value and create a list of application types to be replaced
# use the variable name `application_types_to_replace`
#  YOUR CODE GOES HERE
application_types_to_replace = apptype.where(apptype<100).dropna().index.to_list()
print(application_types_to_replace)

['T13', 'T12', 'T2', 'T25', 'T14', 'T29', 'T15', 'T17']


In [ ]:
#create list of apptypes to delete
application_types_to_delete2 = apptype.where(apptype<10).dropna().index.to_list()
print(application_types_to_delete2)

#create list of apptypes to bin as other

application_types_to_replace2 = apptype.where(apptype<100).dropna().index.to_list()
print(application_types_to_replace2)

['T25', 'T14', 'T29', 'T15', 'T17']
['T13', 'T12', 'T2', 'T25', 'T14', 'T29', 'T15', 'T17']


In [ ]:
# create copy of original df to use for Model3 binning
application_df2 = application_df.copy()
application_df2

APPLICATION_TYPE       AFFILIATION CLASSIFICATION      USE_CASE  \
0                  T10       Independent          C1000    ProductDev   
1                   T3       Independent          C2000  Preservation   
2                   T5  CompanySponsored          C3000    ProductDev   
3                   T3  CompanySponsored          C2000  Preservation   
4                   T3       Independent          C1000     Heathcare   
...                ...               ...            ...           ...   
34294               T4       Independent          C1000    ProductDev   
34295               T4  CompanySponsored          C3000    ProductDev   
34296               T3  CompanySponsored          C2000  Preservation   
34297               T5       Independent          C3000    ProductDev   
34298               T3       Independent          C1000  Preservation   

       ORGANIZATION  STATUS     INCOME_AMT SPECIAL_CONSIDERATIONS   ASK_AMT  \
0       Association       1              0                      N      5000   
1      Co-operative       1         1-9999                      N    108590   
2       Association       1              0                      N      5000   
3             Trust       1    10000-24999                      N      6692   
4             Trust       1  100000-499999                      N    142590   
...             ...     ...            ...                    ...       ...   
34294   Association       1              0                      N      5000   
34295   Association       1              0                      N      5000   
34296   Association       1              0                      N      5000   
34297   Association       1              0                      N      5000   
34298  Co-operative       1          1M-5M                      N  36500179   

       IS_SUCCESSFUL  
0                  1  
1                  1  
2                  0  
3                  1  
4                  1  
...              ...  
34294              0  
34295              0  
34296              0  
34297              1  
34298              0  

[34299 rows x 10 columns]

In [ ]:
# Replace in dataframe
for app in application_types_to_replace:
    application_df['APPLICATION_TYPE'] = application_df['APPLICATION_TYPE'].replace(app,"Other")

# Check to make sure binning was successful
application_df['APPLICATION_TYPE'].value_counts()

T3       27037
T4        1542
T6        1216
T5        1173
T19       1065
T8         737
T7         725
T10        528
T9         156
Other      120
Name: APPLICATION_TYPE, dtype: int64

In [ ]:
# Drop and Replace in dataframe
for app in application_types_to_delete2:
    application_df2.drop(application_df2[application_df2['APPLICATION_TYPE'] == app].index, inplace = True)

for app in application_types_to_replace2:
    application_df2['APPLICATION_TYPE'] = application_df2['APPLICATION_TYPE'].replace(app,"Other")

# Check to make sure binning was successful
application_df2['APPLICATION_TYPE'].value_counts()

T3       27037
T4        1542
T6        1216
T5        1173
T19       1065
T8         737
T7         725
T10        528
T9         156
Other      109
Name: APPLICATION_TYPE, dtype: int64

In [ ]:
# Look at CLASSIFICATION value counts for binning
#  YOUR CODE GOES HERE
classifytype = application_df['CLASSIFICATION'].value_counts()
classifytype
# bin classifytype with counts <200 - would create 8 bins
#helper text indicates binning with counts <1000 ending up with 6 bins

C1000    17326
C2000     6074
C1200     4837
C3000     1918
C2100     1883
         ...  
C4120        1
C8210        1
C2561        1
C4500        1
C2150        1
Name: CLASSIFICATION, Length: 71, dtype: int64

In [ ]:
# You may find it helpful to look at CLASSIFICATION value counts >1
#  YOUR CODE GOES HERE
cls = classifytype.where(classifytype>1)
cls.dropna(inplace = True)
cls

C1000    17326.0
C2000     6074.0
C1200     4837.0
C3000     1918.0
C2100     1883.0
C7000      777.0
C1700      287.0
C4000      194.0
C5000      116.0
C1270      114.0
C2700      104.0
C2800       95.0
C7100       75.0
C1300       58.0
C1280       50.0
C1230       36.0
C1400       34.0
C7200       32.0
C2300       32.0
C1240       30.0
C8000       20.0
C7120       18.0
C1500       16.0
C1800       15.0
C6000       15.0
C1250       14.0
C8200       11.0
C1238       10.0
C1278       10.0
C1235        9.0
C1237        9.0
C7210        7.0
C2400        6.0
C1720        6.0
C4100        6.0
C1257        5.0
C1600        5.0
C1260        3.0
C2710        3.0
C0           3.0
C3200        2.0
C1234        2.0
C1246        2.0
C1267        2.0
C1256        2.0
Name: CLASSIFICATION, dtype: float64

In [ ]:
# Choose a cutoff value and create a list of classifications to be replaced
# use the variable name `classifications_to_replace`
#  YOUR CODE GOES HERE
classifications_to_replace = classifytype.where(classifytype<200).dropna().index.to_list()
print(classifications_to_replace)

['C4000', 'C5000', 'C1270', 'C2700', 'C2800', 'C7100', 'C1300', 'C1280', 'C1230', 'C1400', 'C7200', 'C2300', 'C1240', 'C8000', 'C7120', 'C1500', 'C1800', 'C6000', 'C1250', 'C8200', 'C1238', 'C1278', 'C1235', 'C1237', 'C7210', 'C2400', 'C1720', 'C4100', 'C1257', 'C1600', 'C1260', 'C2710', 'C0', 'C3200', 'C1234', 'C1246', 'C1267', 'C1256', 'C2190', 'C4200', 'C2600', 'C5200', 'C1370', 'C1248', 'C6100', 'C1820', 'C1900', 'C1236', 'C3700', 'C2570', 'C1580', 'C1245', 'C2500', 'C1570', 'C1283', 'C2380', 'C1732', 'C1728', 'C2170', 'C4120', 'C8210', 'C2561', 'C4500', 'C2150']


In [ ]:
#create list of apptypes to delete
classifications_to_delete2 = classifytype.where(classifytype<2).dropna().index.to_list()
print(classifications_to_delete2)

#create list of apptypes to bin as other

classifications_to_replace2 = classifytype.where(classifytype<200).dropna().index.to_list()
print(classifications_to_replace2)

['C2190', 'C4200', 'C2600', 'C5200', 'C1370', 'C1248', 'C6100', 'C1820', 'C1900', 'C1236', 'C3700', 'C2570', 'C1580', 'C1245', 'C2500', 'C1570', 'C1283', 'C2380', 'C1732', 'C1728', 'C2170', 'C4120', 'C8210', 'C2561', 'C4500', 'C2150']
['C4000', 'C5000', 'C1270', 'C2700', 'C2800', 'C7100', 'C1300', 'C1280', 'C1230', 'C1400', 'C7200', 'C2300', 'C1240', 'C8000', 'C7120', 'C1500', 'C1800', 'C6000', 'C1250', 'C8200', 'C1238', 'C1278', 'C1235', 'C1237', 'C7210', 'C2400', 'C1720', 'C4100', 'C1257', 'C1600', 'C1260', 'C2710', 'C0', 'C3200', 'C1234', 'C1246', 'C1267', 'C1256', 'C2190', 'C4200', 'C2600', 'C5200', 'C1370', 'C1248', 'C6100', 'C1820', 'C1900', 'C1236', 'C3700', 'C2570', 'C1580', 'C1245', 'C2500', 'C1570', 'C1283', 'C2380', 'C1732', 'C1728', 'C2170', 'C4120', 'C8210', 'C2561', 'C4500', 'C2150']


In [ ]:
# Replace in dataframe
for cls in classifications_to_replace:
    application_df['CLASSIFICATION'] = application_df['CLASSIFICATION'].replace(cls,"Other")
    
# Check to make sure binning was successful
application_df['CLASSIFICATION'].value_counts()

C1000    17326
C2000     6074
C1200     4837
C3000     1918
C2100     1883
Other     1197
C7000      777
C1700      287
Name: CLASSIFICATION, dtype: int64

In [ ]:
# Drop and Replace in dataframe
for cls in classifications_to_delete2:
    application_df2.drop(application_df2[application_df2['CLASSIFICATION'] == cls].index, inplace = True)

for cls in classifications_to_replace:
    application_df2['CLASSIFICATION'] = application_df2['CLASSIFICATION'].replace(cls,"Other")
    
# Check to make sure binning was successful
application_df2['CLASSIFICATION'].value_counts()


C1000    17315
C2000     6074
C1200     4837
C3000     1918
C2100     1883
Other     1171
C7000      777
C1700      287
Name: CLASSIFICATION, dtype: int64

In [ ]:
application_df.columns

Index(['APPLICATION_TYPE', 'AFFILIATION', 'CLASSIFICATION', 'USE_CASE',
       'ORGANIZATION', 'STATUS', 'INCOME_AMT', 'SPECIAL_CONSIDERATIONS',
       'ASK_AMT', 'IS_SUCCESSFUL'],
      dtype='object')

In [ ]:
application_df2.columns

Index(['APPLICATION_TYPE', 'AFFILIATION', 'CLASSIFICATION', 'USE_CASE',
       'ORGANIZATION', 'STATUS', 'INCOME_AMT', 'SPECIAL_CONSIDERATIONS',
       'ASK_AMT', 'IS_SUCCESSFUL'],
      dtype='object')

In [ ]:
application_df

APPLICATION_TYPE       AFFILIATION CLASSIFICATION      USE_CASE  \
0                  T10       Independent          C1000    ProductDev   
1                   T3       Independent          C2000  Preservation   
2                   T5  CompanySponsored          C3000    ProductDev   
3                   T3  CompanySponsored          C2000  Preservation   
4                   T3       Independent          C1000     Heathcare   
...                ...               ...            ...           ...   
34294               T4       Independent          C1000    ProductDev   
34295               T4  CompanySponsored          C3000    ProductDev   
34296               T3  CompanySponsored          C2000  Preservation   
34297               T5       Independent          C3000    ProductDev   
34298               T3       Independent          C1000  Preservation   

       ORGANIZATION  STATUS     INCOME_AMT SPECIAL_CONSIDERATIONS   ASK_AMT  \
0       Association       1              0                      N      5000   
1      Co-operative       1         1-9999                      N    108590   
2       Association       1              0                      N      5000   
3             Trust       1    10000-24999                      N      6692   
4             Trust       1  100000-499999                      N    142590   
...             ...     ...            ...                    ...       ...   
34294   Association       1              0                      N      5000   
34295   Association       1              0                      N      5000   
34296   Association       1              0                      N      5000   
34297   Association       1              0                      N      5000   
34298  Co-operative       1          1M-5M                      N  36500179   

       IS_SUCCESSFUL  
0                  1  
1                  1  
2                  0  
3                  1  
4                  1  
...              ...  
34294              0  
34295              0  
34296              0  
34297              1  
34298              0  

[34299 rows x 10 columns]

In [ ]:
application_df2

APPLICATION_TYPE       AFFILIATION CLASSIFICATION      USE_CASE  \
0                  T10       Independent          C1000    ProductDev   
1                   T3       Independent          C2000  Preservation   
2                   T5  CompanySponsored          C3000    ProductDev   
3                   T3  CompanySponsored          C2000  Preservation   
4                   T3       Independent          C1000     Heathcare   
...                ...               ...            ...           ...   
34294               T4       Independent          C1000    ProductDev   
34295               T4  CompanySponsored          C3000    ProductDev   
34296               T3  CompanySponsored          C2000  Preservation   
34297               T5       Independent          C3000    ProductDev   
34298               T3       Independent          C1000  Preservation   

       ORGANIZATION  STATUS     INCOME_AMT SPECIAL_CONSIDERATIONS   ASK_AMT  \
0       Association       1              0                      N      5000   
1      Co-operative       1         1-9999                      N    108590   
2       Association       1              0                      N      5000   
3             Trust       1    10000-24999                      N      6692   
4             Trust       1  100000-499999                      N    142590   
...             ...     ...            ...                    ...       ...   
34294   Association       1              0                      N      5000   
34295   Association       1              0                      N      5000   
34296   Association       1              0                      N      5000   
34297   Association       1              0                      N      5000   
34298  Co-operative       1          1M-5M                      N  36500179   

       IS_SUCCESSFUL  
0                  1  
1                  1  
2                  0  
3                  1  
4                  1  
...              ...  
34294              0  
34295              0  
34296              0  
34297              1  
34298              0  

[34262 rows x 10 columns]

In [ ]:
# Convert categorical data to numeric with `pd.get_dummies`
#  YOUR CODE GOES HERE
dummies = pd.get_dummies(application_df)
dummies

STATUS   ASK_AMT  IS_SUCCESSFUL  APPLICATION_TYPE_Other  \
0           1      5000              1                       0   
1           1    108590              1                       0   
2           1      5000              0                       0   
3           1      6692              1                       0   
4           1    142590              1                       0   
...       ...       ...            ...                     ...   
34294       1      5000              0                       0   
34295       1      5000              0                       0   
34296       1      5000              0                       0   
34297       1      5000              1                       0   
34298       1  36500179              0                       0   

       APPLICATION_TYPE_T10  APPLICATION_TYPE_T19  APPLICATION_TYPE_T3  \
0                         1                     0                    0   
1                         0                     0                    1   
2                         0                     0                    0   
3                         0                     0                    1   
4                         0                     0                    1   
...                     ...                   ...                  ...   
34294                     0                     0                    0   
34295                     0                     0                    0   
34296                     0                     0                    1   
34297                     0                     0                    0   
34298                     0                     0                    1   

       APPLICATION_TYPE_T4  APPLICATION_TYPE_T5  APPLICATION_TYPE_T6  ...  \
0                        0                    0                    0  ...   
1                        0                    0                    0  ...   
2                        0                    1                    0  ...   
3                        0                    0                    0  ...   
4                        0                    0                    0  ...   
...                    ...                  ...                  ...  ...   
34294                    1                    0                    0  ...   
34295                    1                    0                    0  ...   
34296                    0                    0                    0  ...   
34297                    0                    1                    0  ...   
34298                    0                    0                    0  ...   

       INCOME_AMT_1-9999  INCOME_AMT_10000-24999  INCOME_AMT_100000-499999  \
0                      0                       0                         0   
1                      1                       0                         0   
2                      0                       0                         0   
3                      0                       1                         0   
4                      0                       0                         1   
...                  ...                     ...                       ...   
34294                  0                       0                         0   
34295                  0                       0                         0   
34296                  0                       0                         0   
34297                  0                       0                         0   
34298                  0                       0                         0   

       INCOME_AMT_10M-50M  INCOME_AMT_1M-5M  INCOME_AMT_25000-99999  \
0                       0                 0                       0   
1                       0                 0                       0   
2                       0                 0                       0   
3                       0                 0                       0   
4                       0                 0                       0   
...                   ...            

In [ ]:
dummies2 = pd.get_dummies(application_df2)
dummies2

STATUS   ASK_AMT  IS_SUCCESSFUL  APPLICATION_TYPE_Other  \
0           1      5000              1                       0   
1           1    108590              1                       0   
2           1      5000              0                       0   
3           1      6692              1                       0   
4           1    142590              1                       0   
...       ...       ...            ...                     ...   
34294       1      5000              0                       0   
34295       1      5000              0                       0   
34296       1      5000              0                       0   
34297       1      5000              1                       0   
34298       1  36500179              0                       0   

       APPLICATION_TYPE_T10  APPLICATION_TYPE_T19  APPLICATION_TYPE_T3  \
0                         1                     0                    0   
1                         0                     0                    1   
2                         0                     0                    0   
3                         0                     0                    1   
4                         0                     0                    1   
...                     ...                   ...                  ...   
34294                     0                     0                    0   
34295                     0                     0                    0   
34296                     0                     0                    1   
34297                     0                     0                    0   
34298                     0                     0                    1   

       APPLICATION_TYPE_T4  APPLICATION_TYPE_T5  APPLICATION_TYPE_T6  ...  \
0                        0                    0                    0  ...   
1                        0                    0                    0  ...   
2                        0                    1                    0  ...   
3                        0                    0                    0  ...   
4                        0                    0                    0  ...   
...                    ...                  ...                  ...  ...   
34294                    1                    0                    0  ...   
34295                    1                    0                    0  ...   
34296                    0                    0                    0  ...   
34297                    0                    1                    0  ...   
34298                    0                    0                    0  ...   

       INCOME_AMT_1-9999  INCOME_AMT_10000-24999  INCOME_AMT_100000-499999  \
0                      0                       0                         0   
1                      1                       0                         0   
2                      0                       0                         0   
3                      0                       1                         0   
4                      0                       0                         1   
...                  ...                     ...                       ...   
34294                  0                       0                         0   
34295                  0                       0                         0   
34296                  0                       0                         0   
34297                  0                       0                         0   
34298                  0                       0                         0   

       INCOME_AMT_10M-50M  INCOME_AMT_1M-5M  INCOME_AMT_25000-99999  \
0                       0                 0                       0   
1                       0                 0                       0   
2                       0                 0                       0   
3                       0                 0                       0   
4                       0                 0                       0   
...                   ...            

In [ ]:
dummies.columns

Index(['STATUS', 'ASK_AMT', 'IS_SUCCESSFUL', 'APPLICATION_TYPE_Other',
       'APPLICATION_TYPE_T10', 'APPLICATION_TYPE_T19', 'APPLICATION_TYPE_T3',
       'APPLICATION_TYPE_T4', 'APPLICATION_TYPE_T5', 'APPLICATION_TYPE_T6',
       'APPLICATION_TYPE_T7', 'APPLICATION_TYPE_T8', 'APPLICATION_TYPE_T9',
       'AFFILIATION_CompanySponsored', 'AFFILIATION_Family/Parent',
       'AFFILIATION_Independent', 'AFFILIATION_National', 'AFFILIATION_Other',
       'AFFILIATION_Regional', 'CLASSIFICATION_C1000', 'CLASSIFICATION_C1200',
       'CLASSIFICATION_C1700', 'CLASSIFICATION_C2000', 'CLASSIFICATION_C2100',
       'CLASSIFICATION_C3000', 'CLASSIFICATION_C7000', 'CLASSIFICATION_Other',
       'USE_CASE_CommunityServ', 'USE_CASE_Heathcare', 'USE_CASE_Other',
       'USE_CASE_Preservation', 'USE_CASE_ProductDev',
       'ORGANIZATION_Association', 'ORGANIZATION_Co-operative',
       'ORGANIZATION_Corporation', 'ORGANIZATION_Trust', 'INCOME_AMT_0',
       'INCOME_AMT_1-9999', 'INCOME_AMT_10000-2499

In [ ]:
dummies2.columns

Index(['STATUS', 'ASK_AMT', 'IS_SUCCESSFUL', 'APPLICATION_TYPE_Other',
       'APPLICATION_TYPE_T10', 'APPLICATION_TYPE_T19', 'APPLICATION_TYPE_T3',
       'APPLICATION_TYPE_T4', 'APPLICATION_TYPE_T5', 'APPLICATION_TYPE_T6',
       'APPLICATION_TYPE_T7', 'APPLICATION_TYPE_T8', 'APPLICATION_TYPE_T9',
       'AFFILIATION_CompanySponsored', 'AFFILIATION_Family/Parent',
       'AFFILIATION_Independent', 'AFFILIATION_National', 'AFFILIATION_Other',
       'AFFILIATION_Regional', 'CLASSIFICATION_C1000', 'CLASSIFICATION_C1200',
       'CLASSIFICATION_C1700', 'CLASSIFICATION_C2000', 'CLASSIFICATION_C2100',
       'CLASSIFICATION_C3000', 'CLASSIFICATION_C7000', 'CLASSIFICATION_Other',
       'USE_CASE_CommunityServ', 'USE_CASE_Heathcare', 'USE_CASE_Other',
       'USE_CASE_Preservation', 'USE_CASE_ProductDev',
       'ORGANIZATION_Association', 'ORGANIZATION_Co-operative',
       'ORGANIZATION_Corporation', 'ORGANIZATION_Trust', 'INCOME_AMT_0',
       'INCOME_AMT_1-9999', 'INCOME_AMT_10000-2499

In [ ]:
# Split our preprocessed data into our features and target arrays
#  YOUR CODE GOES HERE
y = dummies["IS_SUCCESSFUL"].values
X = dummies.drop(["IS_SUCCESSFUL"],1).values

# Split the preprocessed data into a training and testing dataset
#  YOUR CODE GOES HERE
X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=42)

<ipython-input-32-560a279ad4bf>:4: FutureWarning: In a future version of pandas all arguments of DataFrame.drop except for the argument 'labels' will be keyword-only
  X = dummies.drop(["IS_SUCCESSFUL"],1).values


In [ ]:
# Split our preprocessed data into our features and target arrays
#  YOUR CODE GOES HERE
y2 = dummies2["IS_SUCCESSFUL"].values
X2 = dummies2.drop(["IS_SUCCESSFUL"],1).values

# Split the preprocessed data into a training and testing dataset
#  YOUR CODE GOES HERE
X2_train, X2_test, y2_train, y2_test = train_test_split(X2, y2, random_state=42)

<ipython-input-33-d50f3e192e95>:4: FutureWarning: In a future version of pandas all arguments of DataFrame.drop except for the argument 'labels' will be keyword-only
  X2 = dummies2.drop(["IS_SUCCESSFUL"],1).values


In [ ]:
# Create a StandardScaler instances
scaler = StandardScaler()

# Fit the StandardScaler
X_scaler = scaler.fit(X_train)

# Scale the data
X_train_scaled = X_scaler.transform(X_train)
X_test_scaled = X_scaler.transform(X_test)

In [ ]:
# Create a StandardScaler instances
scaler2 = StandardScaler()

# Fit the StandardScaler
X2_scaler = scaler2.fit(X2_train)

# Scale the data
X2_train_scaled = X2_scaler.transform(X2_train)
X2_test_scaled = X2_scaler.transform(X2_test)

## Compile, Train and Evaluate the Model

Initial Model two layers relu/relu 8/5

In [ ]:
# Define the model - deep neural net, i.e., the number of input features and hidden nodes for each layer.
#  YOUR CODE GOES HERE  (add class activity reference here, pulled from ClassStarterCode.ipynb saved to this resources directory)
number_input_features = len(X_train[0])
hidden_nodes_layer1 =  8
hidden_nodes_layer2 = 5

nn = tf.keras.models.Sequential()

# First hidden layer
#  YOUR CODE GOES HERE
nn.add(
    tf.keras.layers.Dense(units=hidden_nodes_layer1, input_dim=number_input_features, activation="relu")
)

# Second hidden layer
#  YOUR CODE GOES HERE
nn.add(tf.keras.layers.Dense(units=hidden_nodes_layer2, activation="relu"))

# Output layer
#  YOUR CODE GOES HERE
nn.add(tf.keras.layers.Dense(units=1, activation="sigmoid"))

# Check the structure of the model
nn.summary()

Model: "sequential"
_________________________________________________________________
 Layer (type)                Output Shape              Param #   
 dense (Dense)               (None, 8)                 376       
                                                                 
 dense_1 (Dense)             (None, 5)                 45        
                                                                 
 dense_2 (Dense)             (None, 1)                 6         
                                                                 
Total params: 427
Trainable params: 427
Non-trainable params: 0
_________________________________________________________________


In [ ]:
# Compile the model
#  YOUR CODE GOES HERE
nn.compile(loss="binary_crossentropy", optimizer="adam", metrics=["accuracy"])

In [ ]:
# Train the model anc create a callback that saves the model's weights every five epochs
# https://keras.io/api/callbacks/
# https://keras.io/api/callbacks/model_checkpoint/
checkpoint_filepath = '/Resources/checkpoint-1'

my_callbacks = tf.keras.callbacks.ModelCheckpoint(    
    filepath=checkpoint_filepath,
    save_weights_only=True,
    monitor='val_accuracy',
    mode='auto',
    save_freq = 5)


fit_model = nn.fit(X_train_scaled,y_train,epochs=100, callbacks=my_callbacks)

Epoch 1/100
804/804 [==============================] - 18s 18ms/step - loss: 0.6471 - accuracy: 0.6579
Epoch 2/100
804/804 [==============================] - 12s 15ms/step - loss: 0.5734 - accuracy: 0.7194
Epoch 3/100
804/804 [==============================] - 5s 7ms/step - loss: 0.5647 - accuracy: 0.7233
Epoch 4/100
804/804 [==============================] - 6s 7ms/step - loss: 0.5605 - accuracy: 0.7241
Epoch 5/100
804/804 [==============================] - 6s 8ms/step - loss: 0.5574 - accuracy: 0.7260
Epoch 6/100
804/804 [==============================] - 5s 7ms/step - loss: 0.5557 - accuracy: 0.7264
Epoch 7/100
804/804 [==============================] - 6s 8ms/step - loss: 0.5540 - accuracy: 0.7269
Epoch 8/100
804/804 [==============================] - 5s 7ms/step - loss: 0.5528 - accuracy: 0.7284
Epoch 9/100
804/804 [==============================] - 7s 8ms/step - loss: 0.5523 - accuracy: 0.7284
Epoch 10/100
804/804 [==============================] - 6s 7ms/step - loss: 0.5518 - ac

In [ ]:
# Evaluate the model using the test data
model_loss, model_accuracy = nn.evaluate(X_test_scaled,y_test,verbose=2)
print(f"Loss: {model_loss}, Accuracy: {model_accuracy}")

# # Export our model to HDF5 file
#  YOUR CODE GOES HERE
nn.save('DeepLearningModelnn.h5')

268/268 - 1s - loss: 0.5543 - accuracy: 0.7287 - 745ms/epoch - 3ms/step
Loss: 0.5543107986450195, Accuracy: 0.7287463545799255


MODEL A - more layers, more neurons relu/tanh/relu, 20/20/10

In [ ]:
# Define the model - deep neural net, i.e., the number of input features and hidden nodes for each layer.
#  YOUR CODE GOES HERE  (add class activity reference here, pulled from ClassStarterCode.ipynb saved to this resources directory)
number_input_featuresA = len(X_train[0])
hidden_nodes_layer1A =  20
hidden_nodes_layer2A = 20
hidden_nodes_layer3A =10

nnA = tf.keras.models.Sequential()

# First hidden layer
#  YOUR CODE GOES HERE
nnA.add(
    tf.keras.layers.Dense(units=hidden_nodes_layer1A, input_dim=number_input_featuresA, activation="relu")
)

# Second hidden layer
#  YOUR CODE GOES HERE
nnA.add(tf.keras.layers.Dense(units=hidden_nodes_layer2A, activation="tanh"))

# Third hidden layer
#  YOUR CODE GOES HERE
nnA.add(tf.keras.layers.Dense(units=hidden_nodes_layer3A, activation="relu"))

# Output layer
#  YOUR CODE GOES HERE
nnA.add(tf.keras.layers.Dense(units=1, activation="sigmoid"))

# Check the structure of the model
nnA.summary()

Model: "sequential_1"
_________________________________________________________________
 Layer (type)                Output Shape              Param #   
 dense_3 (Dense)             (None, 20)                940       
                                                                 
 dense_4 (Dense)             (None, 20)                420       
                                                                 
 dense_5 (Dense)             (None, 10)                210       
                                                                 
 dense_6 (Dense)             (None, 1)                 11        
                                                                 
Total params: 1,581
Trainable params: 1,581
Non-trainable params: 0
_________________________________________________________________


In [ ]:
# Compile the model
#  YOUR CODE GOES HERE
nnA.compile(loss="binary_crossentropy", optimizer="adam", metrics=["accuracy"])

In [ ]:
# Train the model anc create a callback that saves the model's weights every five epochs
# https://keras.io/api/callbacks/
# https://keras.io/api/callbacks/model_checkpoint/
checkpoint_filepath = '/Resources/checkpoint'

my_callbacks = tf.keras.callbacks.ModelCheckpoint(    
    filepath=checkpoint_filepath,
    save_weights_only=True,
    monitor='val_accuracy',
    mode='auto',
    save_freq = 5)


fit_model = nnA.fit(X_train_scaled,y_train,epochs=100, callbacks=my_callbacks)

Epoch 1/100
804/804 [==============================] - 10s 10ms/step - loss: 0.5902 - accuracy: 0.6925
Epoch 2/100
804/804 [==============================] - 10s 12ms/step - loss: 0.5609 - accuracy: 0.7278
Epoch 3/100
804/804 [==============================] - 16s 20ms/step - loss: 0.5566 - accuracy: 0.7293
Epoch 4/100
804/804 [==============================] - 8s 10ms/step - loss: 0.5540 - accuracy: 0.7309
Epoch 5/100
804/804 [==============================] - 7s 9ms/step - loss: 0.5532 - accuracy: 0.7308
Epoch 6/100
804/804 [==============================] - 8s 10ms/step - loss: 0.5508 - accuracy: 0.7315
Epoch 7/100
804/804 [==============================] - 8s 10ms/step - loss: 0.5506 - accuracy: 0.7326
Epoch 8/100
804/804 [==============================] - 8s 9ms/step - loss: 0.5496 - accuracy: 0.7328
Epoch 9/100
804/804 [==============================] - 8s 10ms/step - loss: 0.5483 - accuracy: 0.7337
Epoch 10/100
804/804 [==============================] - 7s 9ms/step - loss: 0.548

In [ ]:
# Evaluate the model using the test data
model_loss, model_accuracy = nnA.evaluate(X_test_scaled,y_test,verbose=2)
print(f"Loss: {model_loss}, Accuracy: {model_accuracy}")

# # Export our model to HDF5 file
#  YOUR CODE GOES HERE
nnA.save('DeepLearningModelnnA.h5')


268/268 - 1s - loss: 0.5562 - accuracy: 0.7298 - 584ms/epoch - 2ms/step
Loss: 0.556180477142334, Accuracy: 0.7297959327697754


MODEL B - different bucketing of initial dataset, 2 layers relu/relu 20/20

In [ ]:
# Define the model - deep neural net, i.e., the number of input features and hidden nodes for each layer.
#  YOUR CODE GOES HERE  (add class activity reference here, pulled from ClassStarterCode.ipynb saved to this resources directory)
number_input_featuresB = len(X2_train[0])
hidden_nodes_layer1B =  20
hidden_nodes_layer2B = 20

nnB = tf.keras.models.Sequential()

# First hidden layer
#  YOUR CODE GOES HERE
nnB.add(
    tf.keras.layers.Dense(units=hidden_nodes_layer1B, input_dim=number_input_featuresB, activation="relu")
)

# Second hidden layer
#  YOUR CODE GOES HERE
nnB.add(tf.keras.layers.Dense(units=hidden_nodes_layer2B, activation="relu"))

# Output layer
#  YOUR CODE GOES HERE
nnB.add(tf.keras.layers.Dense(units=1, activation="sigmoid"))

# Check the structure of the model
nnB.summary()

Model: "sequential_2"
_________________________________________________________________
 Layer (type)                Output Shape              Param #   
 dense_7 (Dense)             (None, 20)                940       
                                                                 
 dense_8 (Dense)             (None, 20)                420       
                                                                 
 dense_9 (Dense)             (None, 1)                 21        
                                                                 
Total params: 1,381
Trainable params: 1,381
Non-trainable params: 0
_________________________________________________________________


In [ ]:
# Compile the model
#  YOUR CODE GOES HERE
nnB.compile(loss="binary_crossentropy", optimizer="adam", metrics=["accuracy"])

In [ ]:
# Train the model anc create a callback that saves the model's weights every five epochs
# https://keras.io/api/callbacks/
# https://keras.io/api/callbacks/model_checkpoint/


checkpoint_filepath = '/Resources/checkpointB'
checkpoint_dir = os.path.dirname(checkpoint_filepath)

my_callbacks = tf.keras.callbacks.ModelCheckpoint(    
    filepath=checkpoint_filepath,
    save_weights_only=True,
    monitor='accuracy',
    mode='auto',
    save_freq = 5)


fit_model = nnB.fit(X2_train_scaled,y2_train,epochs=100, callbacks=my_callbacks)

Epoch 1/100
803/803 [==============================] - 6s 7ms/step - loss: 0.5870 - accuracy: 0.7032
Epoch 2/100
803/803 [==============================] - 6s 8ms/step - loss: 0.5594 - accuracy: 0.7240
Epoch 3/100
803/803 [==============================] - 5s 6ms/step - loss: 0.5542 - accuracy: 0.7282
Epoch 4/100
803/803 [==============================] - 6s 7ms/step - loss: 0.5521 - accuracy: 0.7301
Epoch 5/100
803/803 [==============================] - 5s 6ms/step - loss: 0.5508 - accuracy: 0.7291
Epoch 6/100
803/803 [==============================] - 6s 8ms/step - loss: 0.5496 - accuracy: 0.7286
Epoch 7/100
803/803 [==============================] - 5s 6ms/step - loss: 0.5486 - accuracy: 0.7308
Epoch 8/100
803/803 [==============================] - 5s 6ms/step - loss: 0.5478 - accuracy: 0.7302
Epoch 9/100
803/803 [==============================] - 6s 7ms/step - loss: 0.5474 - accuracy: 0.7309
Epoch 10/100
803/803 [==============================] - 5s 6ms/step - loss: 0.5469 - accura

In [ ]:
# Evaluate the model using the test data
model_loss, model_accuracy = nnB.evaluate(X2_test_scaled,y2_test,verbose=2)
print(f"Loss: {model_loss}, Accuracy: {model_accuracy}")

# # Export our model to HDF5 file
#  YOUR CODE GOES HERE
nnB.save('DeepLearningModelnnB.h5')


268/268 - 1s - loss: 0.5500 - accuracy: 0.7259 - 587ms/epoch - 2ms/step
Loss: 0.5499982833862305, Accuracy: 0.7258930802345276


MODEL C five layers all relu, all 100

In [ ]:
# Define the model - deep neural net, i.e., the number of input features and hidden nodes for each layer.
#  YOUR CODE GOES HERE  (add class activity reference here, pulled from ClassStarterCode.ipynb saved to this resources directory)
number_input_featuresC = len(X_train[0])
hidden_nodes_layer1C = 100
hidden_nodes_layer2C = 100
hidden_nodes_layer3C = 100
hidden_nodes_layer4C = 100
hidden_nodes_layer5C = 100

nnC = tf.keras.models.Sequential()

# First hidden layer
#  YOUR CODE GOES HERE
nnC.add(
    tf.keras.layers.Dense(units=hidden_nodes_layer1C, input_dim=number_input_featuresC, activation="relu")
)

# Second hidden layer
#  YOUR CODE GOES HERE
nnC.add(tf.keras.layers.Dense(units=hidden_nodes_layer2C, activation="relu"))

# Third hidden layer
#  YOUR CODE GOES HERE
nnC.add(tf.keras.layers.Dense(units=hidden_nodes_layer3C, activation="relu"))

# Fourth hidden layer
#  YOUR CODE GOES HERE
nnC.add(tf.keras.layers.Dense(units=hidden_nodes_layer4C, activation="relu"))

# Fifth hidden layer
#  YOUR CODE GOES HERE
nnC.add(tf.keras.layers.Dense(units=hidden_nodes_layer5C, activation="relu"))

# Output layer
#  YOUR CODE GOES HERE
nnC.add(tf.keras.layers.Dense(units=1, activation="sigmoid"))

# Check the structure of the model
nnC.summary()

Model: "sequential_3"
_________________________________________________________________
 Layer (type)                Output Shape              Param #   
 dense_10 (Dense)            (None, 100)               4700      
                                                                 
 dense_11 (Dense)            (None, 100)               10100     
                                                                 
 dense_12 (Dense)            (None, 100)               10100     
                                                                 
 dense_13 (Dense)            (None, 100)               10100     
                                                                 
 dense_14 (Dense)            (None, 100)               10100     
                                                                 
 dense_15 (Dense)            (None, 1)                 101       
                                                                 
Total params: 45,201
Trainable params: 45,201
Non-trai

In [ ]:
# Compile the model
#  YOUR CODE GOES HERE
nnC.compile(loss="binary_crossentropy", optimizer="adam", metrics=["accuracy"])

In [ ]:
# Train the model anc create a callback that saves the model's weights every five epochs
# https://keras.io/api/callbacks/
# https://keras.io/api/callbacks/model_checkpoint/
checkpoint_filepath = '/Resources/checkpointC'

my_callbacks = tf.keras.callbacks.ModelCheckpoint(    
    filepath=checkpoint_filepath,
    save_weights_only=True,
    monitor='val_accuracy',
    mode='auto',
    save_freq = 5)


fit_model = nnC.fit(X_train_scaled,y_train,epochs=100, callbacks=my_callbacks)

Epoch 1/100
804/804 [==============================] - 10s 11ms/step - loss: 0.5666 - accuracy: 0.7205
Epoch 2/100
804/804 [==============================] - 9s 11ms/step - loss: 0.5544 - accuracy: 0.7291
Epoch 3/100
804/804 [==============================] - 8s 10ms/step - loss: 0.5510 - accuracy: 0.7313
Epoch 4/100
804/804 [==============================] - 9s 11ms/step - loss: 0.5498 - accuracy: 0.7320
Epoch 5/100
804/804 [==============================] - 9s 11ms/step - loss: 0.5488 - accuracy: 0.7315
Epoch 6/100
804/804 [==============================] - 9s 11ms/step - loss: 0.5469 - accuracy: 0.7331
Epoch 7/100
804/804 [==============================] - 9s 11ms/step - loss: 0.5463 - accuracy: 0.7336
Epoch 8/100
804/804 [==============================] - 9s 11ms/step - loss: 0.5455 - accuracy: 0.7323
Epoch 9/100
804/804 [==============================] - 8s 10ms/step - loss: 0.5451 - accuracy: 0.7325
Epoch 10/100
804/804 [==============================] - 9s 11ms/step - loss: 0.54

In [ ]:
# Evaluate the model using the test data
model_loss, model_accuracy = nnC.evaluate(X_test_scaled,y_test,verbose=2)
print(f"Loss: {model_loss}, Accuracy: {model_accuracy}")

# # Export our model to HDF5 file
#  YOUR CODE GOES HERE
nnC.save('DeepLearningModelnnC.h5')


268/268 - 1s - loss: 0.5863 - accuracy: 0.7324 - 1s/epoch - 4ms/step
Loss: 0.5863363742828369, Accuracy: 0.7323614954948425


ModelD two layers relu/relu, 100/50
last run seemed to peak mid-epoch, overfit?

In [ ]:
# Define the model - deep neural net, i.e., the number of input features and hidden nodes for each layer.
#  YOUR CODE GOES HERE  (add class activity reference here, pulled from ClassStarterCode.ipynb saved to this resources directory)
number_input_featuresD = len(X_train[0])
hidden_nodes_layer1D = 100
hidden_nodes_layer2D = 50

nnD = tf.keras.models.Sequential()

# First hidden layer
#  YOUR CODE GOES HERE
nnD.add(
    tf.keras.layers.Dense(units=hidden_nodes_layer1D, input_dim=number_input_featuresD, activation="relu")
)

# Second hidden layer
#  YOUR CODE GOES HERE
nnD.add(tf.keras.layers.Dense(units=hidden_nodes_layer2D, activation="relu"))


# Output layer
#  YOUR CODE GOES HERE
nnD.add(tf.keras.layers.Dense(units=1, activation="sigmoid"))

# Check the structure of the model
nnD.summary()

Model: "sequential_4"
_________________________________________________________________
 Layer (type)                Output Shape              Param #   
 dense_16 (Dense)            (None, 100)               4700      
                                                                 
 dense_17 (Dense)            (None, 50)                5050      
                                                                 
 dense_18 (Dense)            (None, 1)                 51        
                                                                 
Total params: 9,801
Trainable params: 9,801
Non-trainable params: 0
_________________________________________________________________


In [ ]:
# Compile the model
#  YOUR CODE GOES HERE
nnD.compile(loss="binary_crossentropy", optimizer="adam", metrics=["accuracy"])

In [ ]:
# Train the model
# Train the model anc create a callback that saves the model's weights every five epochs
# https://keras.io/api/callbacks/
# https://keras.io/api/callbacks/model_checkpoint/
checkpoint_filepath = '/Resources/checkpointD'
checkpoint_dir = os.path.dirname(checkpoint_filepath)

batch_size = 32

my_callbacks = tf.keras.callbacks.ModelCheckpoint(    
    filepath=checkpoint_filepath,
    save_weights_only=True,
    verbose = 1,
    monitor='accuracy',
    mode='auto',
    save_freq = 5*batch_size)


fit_model = nnD.fit(X_train_scaled,y_train,epochs=100, callbacks=my_callbacks)

Epoch 1/100
151/804 [====>.........................] - ETA: 2s - loss: 0.5893 - accuracy: 0.7094
Epoch 1: saving model to /Resources/checkpointD
310/804 [==========>...................] - ETA: 1s - loss: 0.5719 - accuracy: 0.7229
Epoch 1: saving model to /Resources/checkpointD
469/804 [================>.............] - ETA: 1s - loss: 0.5705 - accuracy: 0.7223
Epoch 1: saving model to /Resources/checkpointD
626/804 [======================>.......] - ETA: 0s - loss: 0.5697 - accuracy: 0.7224
Epoch 1: saving model to /Resources/checkpointD
796/804 [============================>.] - ETA: 0s - loss: 0.5690 - accuracy: 0.7229
Epoch 1: saving model to /Resources/checkpointD
804/804 [==============================] - 5s 4ms/step - loss: 0.5693 - accuracy: 0.7227
Epoch 2/100
144/804 [====>.........................] - ETA: 2s - loss: 0.5544 - accuracy: 0.7283
Epoch 2: saving model to /Resources/checkpointD
309/804 [==========>...................] - ETA: 2s - loss: 0.5612 - accuracy: 0.7247
Epoc

In [ ]:
# Evaluate the model using the test data
model_loss, model_accuracy = nnD.evaluate(X_test_scaled,y_test,verbose=2)
print(f"Loss: {model_loss}, Accuracy: {model_accuracy}")

268/268 - 1s - loss: 0.5635 - accuracy: 0.7306 - 561ms/epoch - 2ms/step
Loss: 0.563533365726471, Accuracy: 0.7306122183799744


In [ ]:
# # Export our model to HDF5 file
#  YOUR CODE GOES HERE

nnD.save('Resources/DeepLearningModelnnD.h5')

Early trials using different optimizers (not in this nb)\
20,20 neurons 10 epochs relu relu optimizer adam max accuracy 0.7344\
20,20 neurons 10 epochs relu relu optimizer SGD max accuracy 0.7342\
20,20 neurons 10 epochs relu relu optimizer RMSprop max accuracy 0.7342\
6,4 neurons 10 epochs relu relu optimizer RMSprop max accuracy 0.7257\
6,4 neurons 10 epochs relu relu optimizer Ftrl max accuracy 0.5319\
20,20 neurons 10 epochs relu relu optimizer Ftrl max accuracy 0.5318\
12,8 neurons 10 epochs relu relu optimizer Nadam max accuracy 0.7312\
12,8 neurons 10 epochs relu relu optimizer Adadelta max accuracy 0.7323\
2,1 neurons 50 epochs relu relu optimizer Adadelta max accuracy 0.5318\
2,1 neurons 40 epochs relu relu optimizer adam max accuracy 0.7263\
12,8 neurons 10 epochs relu relu optimizer adagrad max accuracy 0.6463\
12,8 neurons 10 epochs relu relu optimizer adamax max accuracy 0.7292\
12,8 neurons 10 epochs relu relu optimizer SGD max accuracy 0.7308\
100,50 neurons 100 epochs tanh tanh optimizer SGD max accuracy 0.7385\
100,50 neurons 100 epochs relu relu optimizer adam max accuracy 0.7411\
100,100,100 neurons 100 epochs relu relu relu optimizer adam max accuracy 0.7416\
20,20,20 neurons 100 epochs relu relu relu optimizer adam max accuracy 0.7394